docs/MATCHMAKING_PROPOSAL.md

```diff
@@ -0,0 +1,171 @@
+This proposal details the specifications for the matchmaking in our app.
+## Overview
+### 1. Queue
+
+The user presses the PLAY button, making him join a queue for the chosen gamemode.
+
+To indicate the server we are queueing for a game, we call the HTTPS endpoint `/queue` with the following parameters:
+
+* `game_mode`: the game mode we are queueing for
+
+*Note: We do not need to provide the user id, as HTTPS cloud functions authenticate the user with their already authenticated firebase identity, which allows to obtain the user uid without explicitly providing it. It also allows to have some security as only authenticated users can call this function. [Source](https://firebase.google.com/docs/functions/callable)*
+### 2. The game - server-side
+
+Once the server receives a request to queue, it acts as follows:
+
+1. Looks for games that are not full and have the same game mode.
```

Comment:
If the point suggested above will be applied then there is also no need for a search because if there is a not full game it can be only one per queue.

Reply:
I think you are confusing queues and games, players and added to games (aka lobbies).

If the game is full we can start the game. The game has the capacity of its game mode.